tests/test_serialization.py

```diff
@@ -136,6 +148,90 @@ def test_pickle_list() -> None:
         fp.seek(0)
         c1 = pickle.load(fp)
         assert c == c1
+        assert get_ref_type(c1) == Optional[List[Any]]
+        assert c1._metadata.element_type is Any
+        assert c1._metadata.optional is True
+
+
+@pytest.mark.parametrize(  # type: ignore
+    "input_,node,element_type,key_type,optional,ref_type",
+    [
+        (GenericList, "list", Any, Any, False, List[Any]),
+        (OptGenericList, "opt_list", Any, Any, True, Optional[List[Any]]),
+        (GenericDict, "dict", Any, Any, False, Dict[Union[str, Enum], Any]),
+        (
+            OptGenericDict,
+            "opt_dict",
+            Any,
+            Any,
+            True,
+            Optional[Dict[Union[str, Enum], Any]],
+        ),
+        (SubscriptedDict, "dict", int, str, False, Dict[str, int]),
+        (SubscriptedList, "list", int, Any, False, List[int]),
+        (
+            DictConfig(content={"a": "foo"}, element_type=str, key_type=str),
+            None,
+            str,
+            str,
+            True,
+            Optional[Dict[str, str]],
+        ),
+        (
+            DictConfig(content={"a": "foo"}, ref_type=Dict[str, str]),
+            None,
+            str,
+            str,
+            True,
+            Optional[Dict[str, str]],
+        ),
+        (
+            ListConfig(content=[1, 2], element_type=int),
+            None,
+            int,
+            Any,
+            True,
+            Optional[List[int]],
+        ),
+        (
+            ListConfig(content=[1, 2], ref_type=List[int]),
+            None,
+            int,
+            Any,
+            True,
+            Optional[List[int]],
+        ),
```

Comment:
Im thinking of removing this test cases since ListConfigs and DictConfigs are never initialised like this.

Reply:
This is emulating how they are initialized with Structured Configs. I think it's fine.

Reply:
Okay, I've modified this tests since ref_type is never passed alone without extracting the key_type and element_type(we get key_type and element_type and then initialise with it). 
I've removed resolve ref_type since it doesn't make sense after all previous PR and can be solved without it and, because of this, I modified tests to get ref_type with `_utils.get_ref_type`.

Reply:
Instead of removing the tests, just pass the additional required information in (key_type and element_type).

Reply:
Yes, I did that but that made duplicated test cases which I removed.